frontend/public/components/catalog-items.jsx

```diff
@@ -13,36 +16,86 @@ export class CatalogTileViewPage extends React.Component {
   constructor(props) {
     super(props);
 
+    const count = _.countBy(props.items, 'kind');
```

Comment:
Doesn't this need to be in `componentDidUpdate` as well?

Reply:
That was a question I raised earlier in this PR.  Do the counts in the filter by Type controls change when a filter is applied or is it always supposed to be the overall totals?

Reply:
It's not just that. If items are added or removed, the counts aren't updated.

Reply:
Ok, can fix.  Do need to know if the counts should reflect filtering or not.

Reply:
I think what you have is fine (not changing the type counts for the filter).

@tlwu2013 let me know if you disagree.

Reply:
Hi @dtaylor113 / @spadgett, I am thinking the filter counts would get updated when users go deeper along the category tree (or text filter is applied). This way, the side panel provides better visibility (to those attributes like "Types", "Certified Level", "Providers") of current filtered offerings. 

See _purple_ boxes:
![filters](https://user-images.githubusercontent.com/5903705/46446150-77142780-c72f-11e8-984f-daed6cc7ac91.png)

Sorry for the delayed response, and thanks for the efforts and tagging me again!

Reply:
Hi @tlwu2013, @spadgett, ok with updating counts when filter is applied.  

irt. show category tabs even when all items filtered out, concerned that users would have to hunt through tabs to find the ones which contain items.   Could we show counts after the vertical tab?  Ex: Languages (6).   And update those counts when filtering, then users would know at a glance which categories do not contain items: 'Languages (0)'.

Reply:
irt updating byType counts when a filter is applied.  Here I applied a Name filter of 'a':
![image](https://user-images.githubusercontent.com/12733153/46479059-3c0c0580-c7bc-11e8-99c6-b9aeed77546d.png)
Which updated counts to show 6 service classes and 3 source-to-image builders.   The issue is if I then select to show just 'a' source-to-image builders, the count for service classes is set to 0 because it is now also being filtered out.  Just want to make sure we are ok with this, seems wonky to me:
![image](https://user-images.githubusercontent.com/12733153/46479146-72498500-c7bc-11e8-91d4-559712e77096.png)

Reply:
> Currently we only show Category tabs which contain items, but if this is too jarring and we want to show all Tabs and show an empty state for Tabs w/o items (due to filtering), then I'm concerned that users would have to hunt through tabs to find the ones which contain items. 

I would say we remove categories and subcategories that have no items at all (regardless of filter), but leave categories that have items that are filtered.

I'm not sure if having to hunt between categories is a problem. I suspect if you're using the filter you either want to filter inside a specific category that you've already selected OR you want to filter all items and don't care about category. When you filter with All Categories selected, it's easy to see what categories have items.

Reply:
It sounds like some more design discussions need to be had around these issues ... @tlwu2013 let's cover this in the Monday meeting so that we can have a broader discussion